node/consensus/time/data_time_reel.go

```diff
@@ -685,15 +685,16 @@ func (d *DataTimeReel) setHead(frame *protobufs.ClockFrame, distance *big.Int) e
 	d.head = frame
 
 	d.headDistance = distance
-	if d.alwaysSend {
+	if d.syncSend {
 		d.newFrameCh <- frame
+	} else {
+		go func() {
+			select {
+			case d.newFrameCh <- frame:
+			default:
+			}
+		}()
 	}
-	go func() {
-		select {
-		case d.newFrameCh <- frame:
```

Comment:
So the trouble is that while we are doing a sync, a new frame may arrive via the PubSub. We can process already that frame while the sync is running. However, since the `newFrameCh` is unbuffered, we won't actually be able to write the new frame (which we are at the same time searching via sync), and we 'lose' this frame that we now have, but didn't process yet. Because of this, we need to wait 20 seconds until the next frame processing interval is kicked again and this frame that we already have is processed.

By making the channel buffered, we will immediately be able to pickup the frame after the sync, and avoid the 20 second delay.